File: js/packages/teams-ai/src/prompts/SystemMessage.ts
/**
 * @module teams-ai
 */
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

import { TemplateSection } from './TemplateSection';

/**
 * A system message.
 */
export class SystemMessage extends TemplateSection {
    /**
     * Creates a new 'SystemMessage' instance.
     * @param template Template to use for this section.
     * @param tokens Optional. Sizing strategy for this section. Defaults to `auto`.
     */
    public constructor(template: string, tokens: number = -1) {
        super(template, 'system', tokens, true, '\n', '');
    }
}

File: python/packages/ai/teams/ai/prompts/system_message.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from __future__ import annotations

from .sections.template_section import TemplateSection


class SystemMessage(TemplateSection):
    """
    A system message
    """

    def __init__(self, template: str, tokens: float = -1):
        """
        Creates a new 'SystemMessage' instance.

        Args:
            template (str): Template to use for this section.
            tokens (int, optional): Sizing strategy for this section. Defaults to `auto`.
        """
        super().__init__(template, "system", tokens, True, "\n", "")
